packages/block-library/src/search/view.js

```diff
@@ -66,6 +80,58 @@ const { actions } = store(
 					actions.closeSearchInput();
 				}
 			},
+			*updateSearch( e ) {
+				const { value } = e.target;
+
+				const ctx = getContext();
+
+				// Don't navigate if the search didn't really change.
+				if ( value === ctx.search ) {
+					return;
+				}
+
+				ctx.search = value;
+
+				// Debounce the search by 300ms to prevent multiple navigations.
+				supersedePreviousSearch?.();
+				const { promise, resolve, reject } = Promise.withResolvers();
```

Comment:
Support for `Promise.withResolvers` is still a bit low (87%) when compared with the rest of the things we use so far in the frontend (>93%). As this is not critical in any sense, let's find viable alternative that doesn't lower down our browser support.

Reply:
Good catch. I'll update this 👍

Reply:
Done in 7b326cee38c577db3c6123895d6ddb8f25089c26